src/session.c

```diff
@@ -439,19 +442,37 @@ struct p11prov_session_pool {
     pthread_mutex_t lock;
 };
 
+static CK_RV token_session_callback(CK_SESSION_HANDLE hSession,
+                                    CK_NOTIFICATION event,
+                                    CK_VOID_PTR pApplication)
+{
+    P11PROV_SESSION *session = (P11PROV_SESSION *)pApplication;
+
+    if (session->session != hSession) {
+        /* something not right, let's ignore this callback */
+        return CKR_OK;
```

Comment:
Not sure CKR_OK is a correct behavior here

Reply:
Returning an error would abort the PKCS#11 function that called the callback. I do not think we want to do that.